### ISSUE TITLE
Remove old CMake code about gcc version check for -frounding-math 

### ISSUE THREAD

**Issue by mglisse** [CGAL/cgal#6923]
https://github.com/CGAL/cgal/blob/b8e342d84525a3ce713e9111ba6d105b6de82c80/Installation/CMakeLists.txt#L463-L466

Does gcc version 10 match this condition?

**Comment by lrineau** [comment]
That is an old code, that should be removed, because it is useless. The actual way `-frounding-math` is added is:
https://github.com/CGAL/cgal/blob/0bd75b7cdb5bc89f39b322caf8f84764134c0a37/Installation/cmake/modules/CGAL_SetupCGALDependencies.cmake#L158-L161

**Comment by lrineau** [comment]
The issue is about Marc worrying that `-frounding-math` would not be correctly added with gcc version 10 or later. And it is not.

There are a lot of unused parts of CMake files that could be removed, but that would require several days of work, and that would be only a cosmetic refactoring, without any change of behavior. That is not my priority for the moment. If anybody is willing to submit a pull-request, I will review it.

**Comment by albert-github** [comment]
I agree with @afabri especially as this is now known so not that difficult to create a pull request.

> There are a lot of unused parts of CMake files that could be removed, but that would require several days of work, and that would be only a cosmetic refactoring, without any change of behavior. That is not my priority for the moment. If anybody is willing to submit a pull-request, I will review it.

This is true but it is also error prone as when the CMake files are changed the engineer will think "what is going on here".

As part of the statement in the quote is:
> ... There are a lot of unused parts of CMake files that could be removed ...

- Is there a list of these know unused parts?
- Is there already an issue to tackle this problem?

Unfortunately (?) I'm lacking the knowledge about the valid versions of the different parts CGAL otherwise I might have had a go at it.

Regarding this specific problem we see that for testing the GCC compiler version there are different type of tests some based on `GCC_VERSION` other on the more general `CMAKE_CXX_COMPILER_VERSION` (sometimes even in the same file e.g. Installation/cmake/modules/CGAL_SetupCGALDependencies.cmake).
Also the test on using the GCC compilers is done in different ways sometimes by `CMAKE_COMPILER_IS_GNUCCX` and on other places by `CMAKE_CXX_COMPILER_ID MATCHES "GNU"`.
So this could be a bit more consistent as well...

**Comment by lrineau** [comment]
I reopen this issue until somebody is willing to create a pull-request to "solve" it.

**Comment by afabri** [comment]
I raised the question, as it sounded as removing a couple of lines.   If it is more complex I agree with Laurent that we have to set priorities. But keeping the issue open and put it in a task list is certainly a better idea than just closing it.

**Comment by lrineau** [comment]
I have modified the title of the issue, to clarify its new intent: remove those old lines of CMake scripts.

Actually, that is a very small subpart of an older issue, unfinished: https://github.com/CGAL/cgal/issues/4815. The TODO item "remove non-header-only support" is the part that we talked about.

The reason why it is a long task is that it will be long (a few days of work) to identify the parts of our CMake scripts that can now be safely removed.


See also related issues/pull-requests: those with the label https://github.com/CGAL/cgal/labels/CMake%20scripts [(click here)](https://github.com/CGAL/cgal/issues?q=is%3Aopen+label%3A%22CMake+scripts%22+sort%3Acreated-desc).
